### ISSUE TITLE
How to import Ref for the TransformTree example from the doc

### ISSUE THREAD

**Issue by mborus** [mahmoud/glom#136]
In the _Transform Tree_ example here
https://glom.readthedocs.io/en/latest/snippets.html#transform-tree

the code uses _Ref_

"With an arbitrary depth tree, _Ref_ can be used to express a recursive spec."

and gives a code example

    etree2tuples = Fill(Ref('ElementTree', (T.tag, Iter(Ref('ElementTree')).all())))

I'm using glom 19.10.0 and it doesn't come with _Ref_

    from glom import Ref 

causes an import error

    ImportError: cannot import name 'Ref' from 'glom' (C:\Python37\lib\site-packages\glom\__init__.py)

I tried finding _Ref_ anywhere in the Glom source but was not successful.

Can you specify how to import it?

**Comment by mahmoud** [comment]
Hi mborus! First off, congrats on the glom talk! Wish I knew German :)

Ref is available on the master branch only right now. You can try it out if you run:

```
$ pip install --upgrade -e git+https://github.com/mahmoud/glom#egg=glom
```

It's lacking in docs and polish right now, but given that it's a very simple (in implementation) type specifier, I'm confident it'll be in the next release, which I can accelerate if it'll help a particular effort. Let me know if you need it published sooner rather than later.

**Comment by mborus** [comment]
> Hi mborus! First off, congrats on the glom talk! Wish I knew German :)

Thanks. Seems like you saw the Cologne Pythoncamp schedule. :) 

It's not really a talk; more like an introduction/training session:  To this yearly event I usually bring something from the real world which I personally like and use myself. So this year, when I had to fight with a lot of JSON and XML, I thought it would be a good idea to show people how to use Glom.

Good to know that `Ref` is coming up. I'm mainly interested to test if I can parse simple XML directly without having to convert it to JSON with other tools. I'll give it a try and see if I get it to work and if I see a problem, I'll post an issue.

>It's lacking in docs and polish right now, but given that it's a very simple (in implementation) type >specifier, I'm confident it'll be in the next release, which I can accelerate if it'll help a particular effort. >Let me know if you need it published sooner rather than later.

Take your time, For next week, I'll just tell people to install it from master like you said if they need this feature right now.

**Comment by mborus** [comment]
Some feedback, using the master branch and the example given,

    etree2tuples2 = Fill(Ref('ElementTree', (T.tag, Iter(Ref('ElementTree')).all())))

returns

    ('html',
      (<generator object Iter._iterate at 0x0000022F7ECFB848>,
       [<Element 'head' at 0x0000022F7ECFE9F8>,
       <Element 'body' at 0x0000022F7ECFE8B8>]))

 instead of

    ('html', [('head', [('title', [])]), ('body', [('p', [])])])

etree2dicts works and returns a _glommable_ json.

**Comment by mahmoud** [comment]
@mborus You're quite right. The problem lies in that `Iter().all()` is really a shortcut that returns `(self, list)`. That construct is presuming `Auto` mode, whereas etree2tuples is using `Fill` mode. 

@kurtbrose are we sure this worked before? I can make `Iter.all()` return `Auto((self, list))`, as a quick fix.

**Comment by kurtbrose** [comment]
oops -- I made some "simple" transformations from ElementTree stuff that was the motivator for creating Ref to generate that example

Auto((self, list)) isn't quite right either, b/c that will change all children

```python
Ref('ElementTree', Fill((T.tag, Auto(Iter(Ref('ElementTree')).all()))))
```
I think would do it...

there's always felt like a "missing piece" from glom is explicit spec versions of tuple, dict, list

this would be a very concrete use case -- so e.g. `Seq(a, b, c)` -- maybe means do `a`, then do `b`, then do `c` -- this could be the explicit spec version of `Auto(tuple)`
